extensions/amp-story/1.0/amp-story.js

```diff
@@ -2772,6 +2804,42 @@ export class AmpStory extends AMP.BaseElement {
         devError(TAG, err, 'Bundle not found for language ' + languageCode);
       });
   }
+
+  /**
+   * @return {number} The index of the last page that should be shown in
+   *     preview mode.
+   * @private
+   */
+  calculateIndexOfLastPageToPreview_() {
+    const minPreviewPages =
+      parseInt(this.viewer_?.getParam('minPreviewPages'), 10) ||
+      DEFAULT_MIN_PAGES_TO_PREVIEW;
+    const pctPagesToPreview =
+      parseInt(this.viewer_?.getParam('pctPagesToPreview'), 10) ||
+      DEFAULT_PCT_PAGES_TO_PREVIEW;
+
+    // We calculate the number of preview pages by taking the larger of the two
+    // values: min # of preview pages vs the % of pages to show.
+    const numPages = this.element.querySelectorAll('amp-story-page').length;
+    let numPreviewPages = Math.ceil((pctPagesToPreview / 100) * numPages);
+    numPreviewPages = Math.max(minPreviewPages, numPreviewPages);
+    numPreviewPages = clamp(numPreviewPages, 1, numPages);
```

Comment:
I might be wrong, but wouldn't this be `const idealPreviewPages = Math.ceil((pctPagesToPreview / 100) * numPages)); const numPreviewPages = clamp(idealPreviewPages, minPreviewPages, numPages)`.

Essentially clamp the ideal number of pages (based on the percentage) between minPreviewPages and numPages.